unit_test/sparse/Test_Sparse_spmv.hpp

```diff
@@ -409,14 +456,19 @@ void test_spmv(lno_t numRows, size_type nnz, lno_t bandwidth,
   for (auto mode : nonTransModes) {
     for (double alpha : testAlphaBeta) {
       for (double beta : testAlphaBeta) {
-        Test::check_spmv(input_mat, input_x, output_y, alpha, beta, mode);
+        mag_t max_error = beta * max_y + alpha * bandwidth * max_val * max_x;
```

Comment:
```suggestion
        mag_t max_error = beta * max_y + alpha * (nnz / nr) * max_val * max_x;
```
Bandwidth is how far nonzeros can be from diagonal, but I think the max error should be based on just entries per row. Or this could use the actual max nz per row, because row_size_variance > 0 means that could be larger than ``nnz/nr``.

Also, it looks like we never call ``test_spmv`` with ``numRows == 0``, but to support that there would need to be a check to avoid divide by zero.

Reply:
Okay, I did not recall what bandwidth meant, I thought it might be a parameter to allow the number of nnz per row to vary within a certain bandwidth. I'll remove that and see how things are working.